### ISSUE TITLE
New widgets

### ISSUE THREAD

**Issue by yesnault** [gizak/termui#5]
Hi,

Have-you got a roadmap with futurs widgets ?

Ball, Triangle, ...

**Comment by gizak** [comment]
Yes, I am going to put up the event system after finishing the grid layout. Then add more widgets like canvas, table, input...

Ball, triangle... not clear what are these about...

**Comment by yesnault** [comment]
Great !
Ball, triangle are not good examples, canvas allows drawing that.

**Comment by cultureulterior** [comment]
Tree structures might be good

**Comment by gizak** [comment]
Hi @cultureulterior, you mean tree, like `tree` command in *Unix?

**Comment by RangerMauve** [comment]
I think @cultureulterior meant something like a directory structure so you can have nested items. Something like [this](http://chenglou.me/react-treeview/)

**Comment by warmans** [comment]
I'd really like a table widget that works more or less like a list but takes a [][]string and renders all the elements in each array in properly sized columns. I think it may be possible to achieve this using: http://golang.org/pkg/text/tabwriter/

e.g. 

```
first    1    100
second   2    200
third    3    300
```

**Comment by cultureulterior** [comment]
gizak, yeah, like pstree or tree, but yes, collapsibility would be good

**Comment by cultureulterior** [comment]
Yes @gizak

**Comment by jwhitlark** [comment]
blessed-contrib has a specialized widget for logs, which would be the first thing I would personally look for.  After that, the table, and possibly the tree.  While sparklines and stuff look cool, the dashboards I've written are 45% logs, 45% tables, and 10% misc., so that's where I'd recommend concentrating initially.
